.github/workflows/deploy_terraform.yml

```diff
@@ -23,15 +23,26 @@ jobs:
       tf_change: ${{ steps.build_vars.outputs.has_terraform_change }}
     steps:
       - name: Check out changes
-        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+        uses: actions/checkout@v4
       - name: Build vars
         id: build_vars
         uses: ./.github/actions/build-vars
 
+  check_for_deprecations:
+    name: Check for Deprecated Resources
+    runs-on: ubuntu-latest
+    needs:
+      - pre_job
+    steps:
+      - name: Check Out Changes
+        uses: actions/checkout@v4
+      - name: Run Deprecation Check
+        run: ./scripts/check_deprecations.sh
```

Comment:
We'll need to add this script to this PR. 
It should probably live in `.github/scripts/tf-deprecations/check_deprecations.sh`